src/styles/calendar.scss

```diff
@@ -0,0 +1,53 @@
+.calendar {
+  width: 7 * ($size + $gap);
+
+  margin-top: 10px;
+  display: flex;
+  flex-wrap: wrap;
+
+  gap: #{$gap};
+
+  @for $i from 28 through 31 {
+    &--month-length--#{$i} {
+      @for $j from 1 through 3 {
+        & :nth-child(#{$i + $j}) {
+          display: none;
+        }
+      }
+    }
+  }
+
+  @for $i from $start-date through $end-date {
+    &__day-#{$i} {
```

Comment:
you really don't need this loop, just remove number of day for each day, and give this styles for `&__day` class